src/test/java/com/vinyl/PlaceOrderTest.java

```diff
@@ -0,0 +1,99 @@
+package com.vinyl;
+
+import com.vinyl.model.Cart;
+import com.vinyl.model.Product;
+import com.vinyl.model.ProductCart;
+import com.vinyl.model.Purchase;
+import com.vinyl.repository.*;
+import org.assertj.core.api.Assertions;
+import org.assertj.core.util.Lists;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.*;
+
+public class PlaceOrderTest extends LoggedInBaseIntegration {
+
+    @Autowired
+    private CartRepository cartRepository;
+
+    @Autowired
+    private PurchaseRepository purchaseRepository;
+
+    @Autowired
+    private ProductCartRepository productCartRepository;
+
+    @Autowired
+    private ProductRepository productRepository;
+
+    @Autowired
+    private PurchaseProductRepository purchaseProductRepository;
+
+    private Cart cart;
+    private Product product;
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        product = defaultEntitiesHelper.createProduct();
+        cart = cartSetup();
+    }
+
+    @Test
+    public void testWhenOk() {
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse();
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+        Purchase purchase = purchaseRepository.findByUser(user);
+        Assertions.assertThat(purchase).isNotNull();
+        Assertions.assertThat(purchaseProductRepository.findByPurchase(purchase).getProduct().getProductName()).isEqualTo(product.getProductName());
```

Comment:
no need to delcare a new method in the repo for this assert, find the purchase by user and check the first product from the product list